system/lib/wasmfs/syscalls.cpp

```diff
@@ -25,6 +25,61 @@ extern "C" {
 
 using namespace wasmfs;
 
+// Copy the file specified by the pathname into JS.
+// Return a pointer to the JS buffer in HEAPU8.
+int emscripten_wasmfs_readFiles(char* path) {
```

Comment:
Why "files" in plural, if it's just one file?

Also, for a C-style API, `_read_file`